backend/src/apiserver/main.go

```diff
@@ -239,6 +282,50 @@ func startHttpProxy(resourceManager *resource.ResourceManager) {
 	glog.Info("Http Proxy started")
 }
 
+func startWebhook(controllerClient ctrlclient.Client, wg *sync.WaitGroup) (*http.Server, error) {
+	glog.Info("Starting the Kubernetes webhooks...")
+
+	tlsCertPath := *webhookTLSCertPath
+	tlsKeyPath := *webhookTLSKeyPath
+
+	topMux := mux.NewRouter()
+
+	pvValidateWebhook, err := webhook.NewPipelineVersionWebhook(controllerClient)
+	if err != nil {
+		return nil, fmt.Errorf("failed to instantiate the Kubernetes webhook: %v", err)
+	}
+
+	topMux.Handle("/webhooks/validate-pipelineversion", pvValidateWebhook)
+
+	server := &http.Server{
+		Addr:    *webhookPortFlag,
+		Handler: topMux,
+	}
+
+	go func() {
+		defer wg.Done()
+
+		if tlsCertPath != "" && tlsKeyPath != "" {
+			if !common.FileExists(tlsCertPath) || !common.FileExists(tlsKeyPath) {
+				glog.Fatalf("TLS certificate/key paths are set but files do not exist")
+			}
+			glog.Info("Starting the Kubernetes webhook with TLS")
+			err := server.ListenAndServeTLS(tlsCertPath, tlsKeyPath)
+			if err != nil && err != http.ErrServerClosed {
```

Comment:
```suggestion
			if err != nil && !errors.Is(err, http.ErrServerClosed) {
```

so that this can succeed against wrapped errors (credit to goland for this one) 
same [here](https://github.com/kubeflow/pipelines/pull/11774/files#diff-d3ae27a3b5003f91b2cbf384575c21f07dd142165daf842fc79fd8459a844829R321)